src/server/game/Chat/ChatCommands/ChatCommandArgs.h

```diff
@@ -99,6 +102,84 @@ struct ArgInfo<std::string, void>
     }
 };
 
+// enum
+template <typename T>
+struct ArgInfo<T, std::enable_if_t<std::is_enum_v<T>>>
+{
+    typedef typename std::underlying_type<T>::type uType;
```

Comment:
lower camel case for typenames feels weird

also
```suggestion
    using U = std::underlying_type_t<T>;
```

Reply:
or hell just move the `using` declaration to just before you actually use it, you only use it in one place and it makes the whole thing way more readable